### PR TITLE
Fix: Selected-cmd bug

### PR DESCRIPTION
**Pull request by jeevithakannan2**
## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Documentation update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Description
- Fixed a bug where floating window appears even though no command is selected.

command

## Testing
- Working as expected.

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Review by adamperkowski — APPROVED** [review]

File: tui/src/state.rs [hunk 1]
```diff
@@ -516,7 +516,11 @@ impl AppState {
 
     pub fn selected_item_is_cmd(&self) -> bool {
         // Any item that is not a directory or up directory (..) must be a command
-        !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
+        if self.selection.selected().is_some() {
+            !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
+        } else {
+            false
+        }
```
Comment (lj3954) [inline tui/src/state.rs:523]:
```suggestion
            self.selection.selected().is_some() && !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
```

**Review by lj3954 — CHANGES_REQUESTED** [review]

File: tui/src/state.rs [hunk 2]
```diff
@@ -516,7 +516,7 @@ impl AppState {
 
     pub fn selected_item_is_cmd(&self) -> bool {
         // Any item that is not a directory or up directory (..) must be a command
-        !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
+            self.selection.selected().is_some() && !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
```
Comment (lj3954) [inline tui/src/state.rs:519]:
```suggestion
        self.selection.selected().is_some() && !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
```

**Review by lj3954 — CHANGES_REQUESTED** [review]

File: tui/src/state.rs [hunk 3]
```diff
@@ -516,7 +516,7 @@ impl AppState {
 
     pub fn selected_item_is_cmd(&self) -> bool {
         // Any item that is not a directory or up directory (..) must be a command
-        !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
+        self.selection.selected().is_some() && !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
```
Comment (lj3954) [inline tui/src/state.rs:519]:
```suggestion
        self.selection.selected().is_some()
            && !(self.selected_item_is_up_dir() || self.selected_item_is_dir())
```

**Review by lj3954 — CHANGES_REQUESTED** [review]

File: tui/src/state.rs (hunk 3 above)
Comment (lj3954) [reply tui/src/state.rs:519]: That'll resolve rustfmt issues.

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**